test/suites/smoke/test-polkadot-decoding.ts

```diff
@@ -7,6 +7,64 @@ import { fail } from "assert";
 
 const pageSize = (process.env.PAGE_SIZE && parseInt(process.env.PAGE_SIZE)) || 500;
 
+const extractStorageKeyComponents = (storageKey: string) => {
+  // The full storage key is composed of
+  // - The 0x prefix (2 characters)
+  // - The module prefix (32 characters)
+  // - The method name (32 characters)
+  // - The parameters (variable length)
+  const moduleKey = storageKey.substring(0, 2 + 32); // Includes the 0x prefix
+  const fnKey = storageKey.substring(2 + 32, 2 + 32 + 32);
+  const paramsKey = storageKey.substring(2 + 32 + 32);
+  return {
+    moduleKey,
+    fnKey,
+    paramsKey,
+  };
+};
+
+// A PRNG used for generating random numbers in a deterministic way
```

Comment:
Random and deterministic in the same phase seem strange.

Can we just simplify it to this?

```js
const randomHex = (nBytes) => [...crypto.getRandomValues(new Uint8Array(nBytes))].map(m=>('0'+m.toString(16)).slice(-2)).join('');
```

Reply:
Ah, now that you mention it, definitely yes. Will do.

FTR, initially, the PRNG had the double purpose of picking a random block height (for another implementation) and generating random hex strings. With the current implementation, it could be simplified to this because we can get away by just logging the storage key that ends up being used.

Reply:
Replaced :+1: